src/modm/platform/dac/sam_x7x/dac.hpp

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright (c) 2023, Christopher Durand
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_SAMX7X_DAC_HPP
+#define MODM_SAMX7X_DAC_HPP
+
+#include <cstdint>
+#include <modm/math/algorithm/prescaler.hpp>
+#include <modm/platform/clock/clockgen.hpp>
+#include <modm/platform/gpio/connector.hpp>
+
+namespace modm::platform
+{
+
+/**
+ * Digital-to-Analog Converter (DACC)
+ *
+ * @author Christopher Durand
+ * @ingroup modm_platform_dac
+ */
+class Dac : public PeripheralDriver
+{
+public:
+	enum class Channel : uint8_t
+	{
+		Channel0 = 0,
+		Channel1 = 1
+	};
+
+	enum class Mode
+	{
+		/// DAC conversion is started by writing data register
+		FreeRunning = 0,
+		/// DAC conversion is started by external trigger event
+		Trigger = 1,
+		/// Continuous conversion at a rate of DAC clock / 12
+		MaxSpeed = 2
+	};
+
+	enum class TriggerSource
+	{
+		Datrg = 0,
+		Tc0 = 1,
+		Tc1 = 2,
+		Tc2 = 3,
+		Pwm0Event0 = 4,
+		Pwm0Event1 = 5,
+		Pwm1Event0 = 6,
+		Pwm1Event1 = 7
+	};
+
+	enum class ChannelStatus : uint32_t
+	{
+		Ch0Enabled = Bit0,
+		Ch1Enabled = Bit1,
+		Ch0Ready = Bit8,
+		Ch1Ready = Bit9
+	};
+	MODM_FLAGS32(ChannelStatus);
+
+	enum class Interrupt : uint32_t
+	{
+		/// Set if the peripheral is ready to accept new data for channel 0.
+		TxReadyCh0 = Bit0,
+		/// Set if the peripheral is ready to accept new data for channel 1.
+		TxReadyCh1 = Bit1,
+		/// A conversion has completed for channel 0. Cleared on read of DACC_ISR.
+		EnfOfConversionCh0 = Bit4,
+		/// A conversion has completed for channel 1. Cleared on read of DACC_ISR.
+		EnfOfConversionCh1 = Bit5
+	};
+	MODM_FLAGS32(Interrupt);
+
+	/// Maximum DAC peripheral clock. The maximum sampling rate is MaxClock / 12.
+	static constexpr auto MaxClock = MHz(12);
+
```

Comment:
```suggestion

	static constexpr size_t Resolution = ((1 << 12) - 1);

```

Reply:
Or rather?
```suggestion

	static constexpr size_t ResolutionBits = 12;

```

Reply:
:+1:, let's have:
```
static constexpr size_t ResolutionBits = 12;
static constexpr size_t MaxOutput = ((1 << ResolutionBits) - 1);
```
Someone could expect `Resolution` to be the bit count.